src/types/contexts/loadingOverlay.ts

```diff
@@ -0,0 +1,16 @@
+export interface LoadingOverlayConfig {
+  key: string;
+  active: boolean;
+}
+
+export interface LoadingOverlayControls {
+  activate: () => void;
+  deactivate: () => void;
+  deregister: () => void;
+}
+
+export interface LoadingOverlayInitializer {
+  register: () => LoadingOverlayControls;
+}
+
+export type LoadingOverlayCtx = LoadingOverlayInitializer | null;
```

Comment:
maybe I'm not seeing something, but why union a context interface with `null` if the context value itself is never null??
```suggestion
export interface LoadingOverlayCtx {
  register: () => LoadingOverlayControls;
}
```

Reply:
The context's default value is null

Reply:
Yes, but why? I'm not a fan of doing a null check every time I want to use a context value, especially because the value will never actually be null so long as the provider is above the consumer's location in the component tree. I stole my preferred solution from an article by kentcdodds or something but it follows this pattern ([example](https://github.com/resident51/resident51/blob/3af1965188dde57c7652495907d33ac05566377a/src/contexts/services/Events.tsx#L30)): 

```typescript
interface MyCtx = { some: 'context stuff'; }

export const MyContext = createContext({} as MyCtx);
```